source/slang/slang-check-expr.cpp

```diff
@@ -1174,6 +1174,23 @@ namespace Slang
             }
         }
 
+        if (auto typePack = as<ConcreteTypePack>(type))
```

Comment:
Yong's note: here we add the front-end logic to derive the differential type of a type pack.
Obviously, a type pack's differential type should be another type pack where each element type is the differential type of the corresponding element in the original type pack. If the original element type is not differentiable, then the corresponding type in the differential type pack is `void`. If none of the types in the original type pack are differentiable, then we treat the entire type pack as non-differnetiable by simply returning `null` here.